File: steps/steps.py

def drawSteps(height):
    rows = height+1
    rCount = 1
    for i in range(rows-1):
        print(" ".rjust(rows-i), end ="")
        print("#"*(rCount+i))


print("Enter a height for your steps ")
height = int(input())
drawSteps(height)

File: steps/steps.rb

def drawSteps(height)
    rows = height + 1
    rCount = 1
    (0..height-1).each do |i|
        print " " *(rows-i)
        print "#" *(rCount+i)
        print "\n"
    end
end

print "Enter Height of Steps "
height = gets.chomp.strip.to_i
drawSteps(height)
